### PR TITLE
SSCS-11336 Get first tier documents for UT

### PR DESCRIPTION
**Pull request by paul-pearson**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-11336

### Change description ###

Get first tier documents for UT

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/bundling/BundlingHandler.java [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+        boolean hasEditedDwpEvidenceBundleDocument = isHasEditedDwpEvidenceBundleDocument(sscsCaseData);
+
+        if (!hasPheRequestOrConfidentialityUnderReview(sscsCaseData, response, hasEditedDwpResponseDocument, hasEditedDwpEvidenceBundleDocument)) {
+
+            sscsCaseData.setMultiBundleConfiguration(getMultiBundleConfigs(sscsCaseData));
+        }
+    }
+
+    private boolean hasPheRequestOrConfidentialityUnderReview(SscsCaseData sscsCaseData, PreSubmitCallbackResponse<SscsCaseData> response, boolean hasEditedDwpResponseDocument, boolean hasEditedDwpEvidenceBundleDocument) {
+        if (!sscsCaseData.isBenefitType(Benefit.CHILD_SUPPORT) && isPheStatusUnderReview(sscsCaseData)
```
Comment (jarekPierchala) [inline src/main/java/uk/gov/hmcts/reform/sscs/bundling/BundlingHandler.java:179]: condition to separate var. Or even too when looking at context

File: src/main/java/uk/gov/hmcts/reform/sscs/bundling/BundlingHandler.java [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+        return !response.getErrors().isEmpty();
+    }
+
+    private boolean isHasEditedDwpEvidenceBundleDocument(SscsCaseData sscsCaseData) {
+        return emptyIfNull(sscsCaseData.getDwpDocuments()).stream()
+                .filter(dwpDocument -> DwpDocumentType.DWP_EVIDENCE_BUNDLE.getValue().equals(dwpDocument.getValue().getDocumentType()))
+                .anyMatch(dwpDocument -> nonNull(dwpDocument.getValue().getEditedDocumentLink()));
+    }
+
+    private boolean isHasEditedDwpResponseDocument(SscsCaseData sscsCaseData) {
```
Comment (jarekPierchala) [inline src/main/java/uk/gov/hmcts/reform/sscs/bundling/BundlingHandler.java:196]: hasEditedDwpResponseDocument

File: src/main/java/uk/gov/hmcts/reform/sscs/bundling/BundlingHandler.java [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+
+    private List<MultiBundleConfig> getUneditedConfigs(SscsCaseData sscsCaseData) {
+        if (sscsCaseData.isLanguagePreferenceWelsh()) {
+            return singletonList(getUnEditedConfigForWelsh());
+        }
+        return singletonList(getUnEditedConfigForEnglish());
+    }
+
+    private List<MultiBundleConfig> getEditedAndUneditedConfigs(SscsCaseData sscsCaseData) {
+        if (sscsCaseData.isLanguagePreferenceWelsh()) {
```
Comment (jarekPierchala) [inline src/main/java/uk/gov/hmcts/reform/sscs/bundling/BundlingHandler.java:231]:
```suggestion
       return sscsCaseData.isLanguagePreferenceWelsh() ? getEditedAndUneditedConfigForWelsh() : getEditedAndUndeditedConfigForEnglish();
```

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/getfirsttierdocuments/GetFirstTierDocumentsAboutToStartEventHandler.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    @Override
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback, String userAuthorisation) {
+        if (!canHandle(callbackType, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        final CaseDetails<SscsCaseData> caseDetails = callback.getCaseDetails();
+        final SscsCaseData sscsCaseData = caseDetails.getCaseData();
+
+        PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(callback.getCaseDetails().getCaseData());
```
Comment (jarekPierchala) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/getfirsttierdocuments/GetFirstTierDocumentsAboutToStartEventHandler.java:45]:
```suggestion
       PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(sscsCaseData);
```

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/getfirsttierdocuments/GetFirstTierDocumentsSubmittedHandler.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+            && callback.getEvent() == EventType.GET_FIRST_TIER_DOCUMENTS
+            && isPostHearingsBEnabled;
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback,
+                                                          String userAuthorisation) {
+
+        SscsCaseData caseData = callback.getCaseDetails().getCaseData();
+
```
Comment (jarekPierchala) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/getfirsttierdocuments/GetFirstTierDocumentsSubmittedHandler.java:44]: no need for empty lines

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/getfirsttierdocuments/GetFirstTierDocumentsSubmittedHandler.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+        Long caseId = Long.valueOf(caseData.getCcdCaseId());
+        CcdCallbackMap callbackMap = BUNDLE_CREATED;
+        log.info("Get first tier documents: handling action {} for case {}", callbackMap, caseId);
```
Comment (jarekPierchala) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/getfirsttierdocuments/GetFirstTierDocumentsSubmittedHandler.java:48]: add empty line before log

**Review by jarekPierchala — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/getfirsttierdocuments/GetFirstTierDocumentsAboutToStartEventHandler.java [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.PreSubmitCallbackHandler;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class GetFirstTierDocumentsAboutToStartEventHandler implements PreSubmitCallbackHandler<SscsCaseData> {
+
+    @Value("${feature.postHearingsB.enabled}")
+    private final boolean isPostHearingsBEnabled;
```
Comment (amoslv1) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/getfirsttierdocuments/GetFirstTierDocumentsAboutToStartEventHandler.java:24]: I would imagine that this needs both flags enabled. In practice, B won't be enabled without A also being enabled, so it's probably obsolete to specify both. Maybe good for peace of mind to have both anyway.

Comment (paul-pearson) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/getfirsttierdocuments/GetFirstTierDocumentsAboutToStartEventHandler.java:27]: Thanks, now checking both flags.

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/getfirsttierdocuments/GetFirstTierDocumentsAboutToStartEventHandlerTest.java [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+    private Callback<SscsCaseData> callback;
+
+    @Mock
+    private CaseDetails<SscsCaseData> caseDetails;
+
+    private SscsCaseData sscsCaseData;
+
+    @BeforeEach
+    public void setUp() {
+        openMocks(this);
```
[the remark this answers is not in the document]
Comment (paul-pearson) [reply src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/getfirsttierdocuments/GetFirstTierDocumentsAboutToStartEventHandlerTest.java]: Thanks, changed accordingly.

**Review by jarekPierchala — DISMISSED** [review]

**Review by amoslv1 — APPROVED** [review]

**Review by jarekPierchala — APPROVED** [review]

**Review by DGordonV1 — APPROVED** [review]

**Review by DGordonV1 — DISMISSED** [review]

**Review by jarekPierchala — APPROVED** [review]

**Pull request merged**